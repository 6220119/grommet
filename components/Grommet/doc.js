"use strict";

exports.__esModule = true;
exports.themeDoc = exports.doc = void 0;

var _reactDesc = require("react-desc");

var _propTypes = require("../../utils/prop-types");

var _mixins = require("../../utils/mixins");

var doc = function doc(Grommet) {
  var DocumentedGrommet = (0, _reactDesc.describe)(Grommet).availableAt((0, _mixins.getAvailableAtBadge)('Grommet', 'Utilities')).description('The top level Grommet container.').usage("import { Grommet } from 'grommet';\n<Grommet>...</Grommet>").intrinsicElement('div');
  DocumentedGrommet.propTypes = {
    background: _propTypes.backgroundDoc,
    dir: _reactDesc.PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.bool, _reactDesc.PropTypes.oneOf(['min'])]).description("Whether to take the whole viewport. 'min' indicates it should\n        take at minimum the whole viewport, allowing the viewport to\n        control scrolling.").defaultValue(false),
    options: _reactDesc.PropTypes.shape({
      layer: _reactDesc.PropTypes.shape({
        singleId: _reactDesc.PropTypes.bool
      })
    }).description("Provides a way to assign a unique id to a single DOM node. Currently, \n        this is only supported for Layer. This prop was created to preserve \n        backwards compatibility with existing behavior by allowing users to \n        opt-in to newer behavior.").defaultValue(undefined),
    plain: _reactDesc.PropTypes.bool.description("Whether or not Grommet should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: _reactDesc.PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: _reactDesc.PropTypes.object.description('Custom styles for Grommet app component.'),
    themeMode: _reactDesc.PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: _reactDesc.PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint."),
    containerTarget: _reactDesc.PropTypes.object.description("The node where Drop and Layer containers are inserted. Defaults to\n      document.body which is almost always the right choice. This is used\n      for less common cases like rendering within an internal node (e.g.\n      shadow root)."),
    messages: _reactDesc.PropTypes.shape({
      format: _reactDesc.PropTypes.func.description("A function that will that will be called with an options object\n        parameter containing the id of the message needed. This function\n        should return the text string for that message. If this format\n        function is not defined, messages will be used from the messages\n        object below or a default message if the id wasn't defined in\n        this object."),
      messages: _reactDesc.PropTypes.shape({
        fileInput: _reactDesc.PropTypes.shape({
          browse: _reactDesc.PropTypes.string,
          dropPrompt: _reactDesc.PropTypes.string,
          dropPromptMultiple: _reactDesc.PropTypes.string,
          files: _reactDesc.PropTypes.string,
          remove: _reactDesc.PropTypes.string,
          removeAll: _reactDesc.PropTypes.string
        }),
        form: _reactDesc.PropTypes.shape({
          invalid: _reactDesc.PropTypes.string,
          required: _reactDesc.PropTypes.string
        }),
        menu: _reactDesc.PropTypes.shape({
          openMenu: _reactDesc.PropTypes.string,
          closeMenu: _reactDesc.PropTypes.string
        }),
        rangeSelector: _reactDesc.PropTypes.shape({
          lower: _reactDesc.PropTypes.string,
          upper: _reactDesc.PropTypes.string
        }),
        select: _reactDesc.PropTypes.shape({
          multiple: _reactDesc.PropTypes.string
        }),
        skipLinks: _reactDesc.PropTypes.shape({
          skipTo: _reactDesc.PropTypes.string
        }),
        tabs: _reactDesc.PropTypes.shape({
          tabContents: _reactDesc.PropTypes.string
        }),
        textInput: _reactDesc.PropTypes.shape({
          enterSelect: _reactDesc.PropTypes.string,
          suggestionsCount: _reactDesc.PropTypes.string,
          suggestionsExist: _reactDesc.PropTypes.string,
          suggestionIsOpen: _reactDesc.PropTypes.string
        }),
        video: _reactDesc.PropTypes.shape({
          closeMenu: _reactDesc.PropTypes.string,
          fullScreen: _reactDesc.PropTypes.string,
          progressMeter: _reactDesc.PropTypes.string,
          scrubber: _reactDesc.PropTypes.string,
          openMenu: _reactDesc.PropTypes.string,
          pauseButton: _reactDesc.PropTypes.string,
          playButton: _reactDesc.PropTypes.string,
          volumeDown: _reactDesc.PropTypes.string,
          volumeUp: _reactDesc.PropTypes.string
        })
      })
    }).description("Custom messages for grommet components. Use this property to\ndefine messages or a function to get localized messages\nfor any grommet children components.").defaultValue({
      messages: {
        fileInput: {
          browse: 'browse',
          dropPrompt: 'Drop file here or',
          dropPromptMultiple: 'Drop files here or',
          files: 'files',
          remove: 'remove',
          removeAll: 'remove all'
        },
        form: {
          invalid: 'invalid',
          required: 'required'
        },
        menu: {
          openMenu: 'Open Menu',
          closeMenu: 'Close Menu'
        },
        rangeSelector: {
          lower: 'Lower Bounds',
          upper: 'Upper Bounds'
        },
        select: {
          multiple: 'multiple'
        },
        skipLinks: {
          skipTo: 'Skip To:'
        },
        tabs: {
          tabContents: 'Tab Contents'
        },
        textInput: {
          enterSelect: '(Press Enter to Select)',
          suggestionsCount: 'suggestions available',
          suggestionsExist: "This input has suggestions use arrow keys\n            to navigate",
          suggestionIsOpen: "Suggestions drop is open, continue to use\n            arrow keys to navigate"
        },
        video: {
          closeMenu: 'close menu',
          fullScreen: 'full screen',
          progressMeter: 'video progress',
          scrubber: 'scrubber',
          openMenu: 'open menu',
          pauseButton: 'pause',
          playButton: 'play',
          volumeDown: 'volume down',
          volumeUp: 'volume up'
        }
      }
    })
  };
  return DocumentedGrommet;
};

exports.doc = doc;
var themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};
exports.themeDoc = themeDoc;