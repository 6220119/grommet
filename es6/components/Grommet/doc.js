import { describe, PropTypes } from 'react-desc';
import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';
export var doc = function doc(Grommet) {
  var DocumentedGrommet = describe(Grommet).availableAt(getAvailableAtBadge('Grommet', 'Utilities')).description('The top level Grommet container.').usage("import { Grommet } from 'grommet';\n<Grommet>...</Grommet>").intrinsicElement('div');
  DocumentedGrommet.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['min'])]).description("Whether to take the whole viewport. 'min' indicates it should\n        take at minimum the whole viewport, allowing the viewport to\n        control scrolling.").defaultValue(false),
    options: PropTypes.shape({
      layer: PropTypes.shape({
        singleId: PropTypes.bool
      })
    }).description("Provides a way to assign a unique id to a single DOM node. Currently, \n        this is only supported for Layer. This prop was created to preserve \n        backwards compatibility with existing behavior by allowing users to \n        opt-in to newer behavior.").defaultValue(undefined),
    plain: PropTypes.bool.description("Whether or not Grommet should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: PropTypes.object.description('Custom styles for Grommet app component.'),
    themeMode: PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint."),
    containerTarget: PropTypes.object.description("The node where Drop and Layer containers are inserted. Defaults to\n      document.body which is almost always the right choice. This is used\n      for less common cases like rendering within an internal node (e.g.\n      shadow root)."),
    messages: PropTypes.shape({
      format: PropTypes.func.description("A function that will that will be called with an options object\n        parameter containing the id of the message needed. This function\n        should return the text string for that message. If this format\n        function is not defined, messages will be used from the messages\n        object below or a default message if the id wasn't defined in\n        this object."),
      messages: PropTypes.shape({
        fileInput: PropTypes.shape({
          browse: PropTypes.string,
          dropPrompt: PropTypes.string,
          dropPromptMultiple: PropTypes.string,
          files: PropTypes.string,
          remove: PropTypes.string,
          removeAll: PropTypes.string
        }),
        form: PropTypes.shape({
          invalid: PropTypes.string,
          required: PropTypes.string
        }),
        menu: PropTypes.shape({
          openMenu: PropTypes.string,
          closeMenu: PropTypes.string
        }),
        rangeSelector: PropTypes.shape({
          lower: PropTypes.string,
          upper: PropTypes.string
        }),
        select: PropTypes.shape({
          multiple: PropTypes.string
        }),
        skipLinks: PropTypes.shape({
          skipTo: PropTypes.string
        }),
        tabs: PropTypes.shape({
          tabContents: PropTypes.string
        }),
        textInput: PropTypes.shape({
          enterSelect: PropTypes.string,
          suggestionsCount: PropTypes.string,
          suggestionsExist: PropTypes.string,
          suggestionIsOpen: PropTypes.string
        }),
        video: PropTypes.shape({
          closeMenu: PropTypes.string,
          fullScreen: PropTypes.string,
          progressMeter: PropTypes.string,
          scrubber: PropTypes.string,
          openMenu: PropTypes.string,
          pauseButton: PropTypes.string,
          playButton: PropTypes.string,
          volumeDown: PropTypes.string,
          volumeUp: PropTypes.string
        })
      })
    }).description("Custom messages for grommet components. Use this property to\ndefine messages or a function to get localized messages\nfor any grommet children components.").defaultValue({
      messages: {
        fileInput: {
          browse: 'browse',
          dropPrompt: 'Drop file here or',
          dropPromptMultiple: 'Drop files here or',
          files: 'files',
          remove: 'remove',
          removeAll: 'remove all'
        },
        form: {
          invalid: 'invalid',
          required: 'required'
        },
        menu: {
          openMenu: 'Open Menu',
          closeMenu: 'Close Menu'
        },
        rangeSelector: {
          lower: 'Lower Bounds',
          upper: 'Upper Bounds'
        },
        select: {
          multiple: 'multiple'
        },
        skipLinks: {
          skipTo: 'Skip To:'
        },
        tabs: {
          tabContents: 'Tab Contents'
        },
        textInput: {
          enterSelect: '(Press Enter to Select)',
          suggestionsCount: 'suggestions available',
          suggestionsExist: "This input has suggestions use arrow keys\n            to navigate",
          suggestionIsOpen: "Suggestions drop is open, continue to use\n            arrow keys to navigate"
        },
        video: {
          closeMenu: 'close menu',
          fullScreen: 'full screen',
          progressMeter: 'video progress',
          scrubber: 'scrubber',
          openMenu: 'open menu',
          pauseButton: 'pause',
          playButton: 'play',
          volumeDown: 'volume down',
          volumeUp: 'volume up'
        }
      }
    })
  };
  return DocumentedGrommet;
};
export var themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};