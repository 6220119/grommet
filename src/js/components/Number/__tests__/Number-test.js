import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { axe } from 'jest-axe';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { Grommet } from '../../Grommet';
import { Number } from '..';

describe('Number', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <Number a11yTitle="test" value={27} />
      </Grommet>,
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('value', () => {
    const component = renderer.create(
      <Grommet>
        <Number value={27} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('units string', () => {
    const component = renderer.create(
      <Grommet>
        <Number value={27} units="tests" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('units object', () => {
    const component = renderer.create(
      <Grommet>
        <Number
          value={27}
          units={{
            label: 'tests',
            size: 'small',
            weight: 'normal',
            color: 'brand',
          }}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('color', () => {
    const component = renderer.create(
      <Grommet>
        <Number value={27} units="tests" color="brand" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('size', () => {
    const component = renderer.create(
      <Grommet>
        {['small', 'medium', 'large', 'xlarge'].map(size => (
          <Number key={size} value={27} units="tests" size={size} />
        ))}
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
