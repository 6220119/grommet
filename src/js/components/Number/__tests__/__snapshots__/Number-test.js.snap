// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number color 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 6px;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  color: #7D4CDB;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 3px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c2"
    >
      tests
    </span>
  </div>
</div>
`;

exports[`Number should have no accessibility violations 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 34px;
  line-height: 40px;
  font-weight: bold;
}

<div
  class="c0"
>
  <span
    aria-label="test"
    class="c1"
  >
    27
  </span>
</div>
`;

exports[`Number size 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 6px;
}

.c2 {
  font-size: 26px;
  line-height: 32px;
  font-weight: bold;
}

.c4 {
  font-size: 34px;
  line-height: 40px;
  font-weight: bold;
}

.c5 {
  font-size: 50px;
  line-height: 56px;
  font-weight: bold;
}

.c6 {
  font-size: 82px;
  line-height: 88px;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 3px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c2"
    >
      tests
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c4"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c4"
    >
      tests
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c5"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c5"
    >
      tests
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c6"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c6"
    >
      tests
    </span>
  </div>
</div>
`;

exports[`Number units object 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 6px;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  font-weight: bold;
}

.c4 {
  font-size: 14px;
  line-height: 20px;
  color: #7D4CDB;
  font-weight: normal;
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 3px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c4"
    >
      tests
    </span>
  </div>
</div>
`;

exports[`Number units string 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 6px;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 3px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      27
    </span>
    <div
      className="c3"
    />
    <span
      className="c2"
    >
      tests
    </span>
  </div>
</div>
`;

exports[`Number value 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 34px;
  line-height: 40px;
  font-weight: bold;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    27
  </span>
</div>
`;
